### DISCUSSION TITLE
Mark certain notifications as read/seen by default

### DISCUSSION THREAD

**Discussion by marckohlbrugge** [excid3/noticed#465] [Q&A] [answered]
I'm using `@user.notifications.unread.count` to show a badge of how many unread notifications a user has. When we show the notifications on the webpage, we mark them all as read.

The problem I run into now is that some special notifications only trigger a delivery, but are never shown on the notifications page. So these notifications should not be counted in `@user.notifications.unread.count`.

Rather than changing the `unread` scope, I figured I'd automatically mark these notifications as read/seen the moment they are created.

How can I can do this?

I tried the code below, but it applies to the `Noticed::Event` rather than the `Noticed::Notification`.

```ruby
class StreakReminderNotifier < ApplicationNotifier
  before_create do
    self.seen_at = Time.zone.now
    self.read_at = Time.zone.now
  end
end
```

**Comment by excid3** [comment, marked as answer]
You can use `notification_methods` to inject code into the notification model.

**Reply by marckohlbrugge** [reply]
Thanks! For some reason I didn't think that would work for a `before_create`block, but it does!

```ruby
class StreakReminderNotifier < ApplicationNotifier
  notification_methods do
    before_create do
      self.seen_at = Time.zone.now
      self.read_at = Time.zone.now
    end
  end
end
```
